### PR TITLE
Product Categories: fixed checkmarks and ghost tableview appearance

### PR DESCRIPTION
**Pull request by pmusolino**
Part of #2000 

As part of the implementation of the Product Categories, I did some small changes and fixes.
- Moved all the classes and xib files under "Edit Categories" folder
- Fixed the checkmark of the product category cell, which was covered by the label when there are tons of subcategories (in my case I used 14 levels!)
- Fixed the appearance of the ghost tableview when categories are loading when dark mode is enabled

## Testing
For this PR just check the code and the CI is enough since the functionality is not completed and new PR will follow.

| Checkmark before (hidden)             |  Checkmarks after (showed) |
:-------------------------:|:-------------------------:
ios 12 issue | ios 12 issue after

| Ghost tableview (before)             |  Ghost tableview (after) |
:-------------------------:|:-------------------------:
Simulator Screen Shot - iPhone 11 Pro - 2020-06-03 at 17 12 41 | Simulator Screen Shot - iPhone 11 Pro - 2020-06-03 at 17 23 13

Update release notes:
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Comment by Ecarrion** [comment]
Thank you for that! :-)

File: WooCommerce/Classes/ViewRelated/Products/Edit Product/Edit Categories/ProductCategoryListViewController.swift [hunk 1]
```diff
@@ -115,7 +115,8 @@ private extension ProductCategoryListViewController {
         let options = GhostOptions(displaysSectionHeader: false,
                                    reuseIdentifier: ProductCategoryTableViewCell.reuseIdentifier,
                                    rowsPerSection: placeholderCategoriesPerSection)
-        ghostTableView.displayGhostContent(options: options)
+        ghostTableView.displayGhostContent(options: options,
+                                           style: .wooDefaultGhostStyle)
```
Comment (jaclync) [inline WooCommerce/Classes/ViewRelated/Products/Edit Product/Edit Categories/ProductCategoryListViewController.swift:119]: nice I didn't know this is the fix for the ghost style! I will apply the same fix for https://github.com/woocommerce/woocommerce-ios/issues/1813 😄

**Review by jaclync — APPROVED** [review]
LGTM! :shipit:

**Review by Garance91540 — APPROVED** [review]
👏 thanks for caring about the details!

**Pull request merged**